aibolit/metrics/fanout/FanOut.py

```diff
@@ -0,0 +1,68 @@
+from aibolit.utils.ast_builder import build_ast
+from aibolit.utils.ast import AST, ASTNodeType
+from itertools import islice
+
+
+class FanOut:
+    '''
+    Fan Out metric is defined as the number of other classes referenced by a class.
+    '''
+    def __init__(self):
+        pass
+
+    def value(self, filename: str) -> int:  # noqa: C901
+
+        # exception are used from https://checkstyle.sourceforge.io/config_metrics.html#ClassFanOutComplexity
+        considered_classes = {'ArrayIndexOutOfBoundsException': 0, 'ArrayList': 0, 'Boolean': 0, 'Byte': 0,
+                              'Character': 0, 'Class': 0, 'Deprecated': 0, 'Deque': 0, 'Double': 0,
+                              'Exception': 0, 'Float': 0, 'FunctionalInterface': 0, 'HashMap': 0,
+                              'HashSet': 0, 'IllegalArgumentException': 0, 'IllegalStateException': 0,
+                              'IndexOutOfBoundsException': 0, 'Integer': 0, 'LinkedList': 0, 'List': 0,
+                              'Long': 0, 'Map': 0, 'NullPointerException': 0, 'Object': 0, 'Override': 0,
+                              'Queue': 0, 'RuntimeException': 0, 'SafeVarargs': 0, 'SecurityException': 0,
+                              'Set': 0, 'Short': 0, 'SortedMap': 0, 'SortedSet': 0, 'String': 0, 'StringBuffer': 0,
+                              'StringBuilder': 0, 'SuppressWarnings': 0, 'Throwable': 0, 'short': 0, 'void': 0,
+                              'TreeMap': 0, 'TreeSet': 0, 'UnsupportedOperationException': 0, 'Void': 0,
+                              'System.out': 0, 'boolean': 0, 'byte': 0, 'char': 0, 'double': 0, 'float': 0,
+                              'int': 0, 'long': 0,
+                              }
+        fan_outs = 0
+
+        # check imported classes
+        tree = AST.build_from_javalang(build_ast(filename))
+        for each_import in (tree.children_with_type(tree.root, ASTNodeType.IMPORT)):
+            name_node, = islice(tree.children_with_type(each_import, ASTNodeType.STRING), 1)
+            new_class = tree.get_attr(name_node, 'string').split('.')[-1]
+            if considered_classes.get(new_class) is None:
+                fan_outs += 1
+                considered_classes[new_class] = 0
+
+        for i in tree.subtrees_with_root_type(ASTNodeType.CLASS_DECLARATION):
+            ast = AST(tree.tree.subgraph(i), i[0])
+            for j in ast.subtrees_with_root_type(ASTNodeType.VARIABLE_DECLARATOR):
+                ast_ = AST(tree.tree.subgraph(j), j[0])
+                for inv_class in ast_.subtrees_with_root_type(ASTNodeType.CLASS_CREATOR):
```

Comment:
`inv_class` sounds like `inverse_class`. We better use here more proper name.

Reply:
Rewrited